cache/edge_lambdas/src/redirector.ts

```diff
@@ -38,14 +38,26 @@ const paramsAreSubset = (
 
 const filterParams = (
   params: URLSearchParams,
-  allow: Set<string>
+  allow: Set<string>,
+  modify?: {
+    [oldParamName: string]: string;
+  }
 ): URLSearchParams => {
   const filtered = new URLSearchParams();
+
   params.forEach((value, key) => {
+    // Check if param is in the forwardParams list
     if (allow.has(key)) {
-      filtered.append(key, value);
+      modify
+        ? // Check if param is also in the modifiedParams list
+          // If the params match, append the new param name to filtered
+          Object.keys(modify).forEach(k => {
+            if (k === key) filtered.append(modify[k], value);
+          })
+        : filtered.append(key, value);
```

Comment:
2 things -
(1) again with side effects, it's not very clear to use a ternary where the result isn't assigned
(2) you don't need to look through the keys of an object and compare them to one you want - accessing a known key is what an object is for!

```suggestion
      filtered.append(modify?.[key] ?? key, value) 
```

or if that seems too 'clever' you could be a bit more verbose and do
```suggestion
if (modify && key in modify) {
  filtered.append(modify[key], value);
} else {
  filtered.append(key, value);
}
```